### DISCUSSION TITLE
External/public listening URL for Metro Bundler

### DISCUSSION THREAD

**Discussion by Stan-RED** [expo/expo-cli#4411] [Q&A]
Hi,

Is it possible to change the listener address for Metro Bundler? I want to make it accessible for some testers that aren't in the local network. For example to have smth like exp://dev.someproject.com. If I simply setup a proxy from external address to internal where Metro is hosted, then firstly I have to generate QR myself and also there are sometimes errors:

```
Cannot connect to Metro.

Try the following to fix the issue:
- Ensure that Metro is running and available on the same network
- Ensure that the Metro URL is correctly set in AppDelegate
```

Thank you!

**Comment by louix** [comment]
Pretty sure this is a built in feature of the Metro dev tools. After running it should spit out a line like this:
```
Developer tools running on http://localhost:19002
```

Visit that in your browser, and the feature should be under the QR code IIRC.

**Reply by Stan-RED** [reply]
@louix, how does this answer my question regarding the possibility to change the default address?

**Reply by louix** [reply]
> I want to make it accessible for some testers that aren't in the local network. For example to have smth like exp://dev.someproject.com.

It addresses this, this is what tunnel mode does. It also generates you the QR code.

**Reply by Stan-RED** [reply]
> It addresses this, this is what tunnel mode does. It also generates you the QR code.

This is close to what I want, but it does not allow me to "change the listener address". It provides me with some tunnel address that I can't control. Can't setup authentication, can't route where I want, etc. For example I have a CI/CD that is building tagged commits and I want to have such versions accessible via exp://{tagname}.dev.someproject.com.
